core/controllers/topics_and_skills_dashboard.py

```diff
@@ -120,37 +120,46 @@ def get(self) -> None:
             for topic_id in classroom['topic_ids']:
                 topic_classroom_dict[topic_id] = classroom['name']
 
-        for topic_summary_dict in topic_summary_dicts:
+        topic_ids = [summary['id'] for summary in topic_summary_dicts]
+        topics = topic_fetchers.get_topics_by_ids(topic_ids)
+        for topic in topics:
+            if topic is None:
+                continue
+            topic_summary_dict = next(
+                topic_summary for topic_summary in topic_summary_dicts
+                if topic_summary['id'] == topic.id)
             topic_summary_dict['classroom'] = topic_classroom_dict.get(
                 topic_summary_dict['id'], None)
-            topic = topic_fetchers.get_topic_by_id(topic_summary_dict['id'])
             upcoming_chapters_count = 0
             overdue_chapters_count = 0
             total_chapters_counts = []
             published_chapters_counts = []
-            for story_reference in topic.canonical_story_references:
-                story = story_fetchers.get_story_by_id(
-                    story_reference.story_id)
+            story_ids = [story_reference.story_id for story_reference in
+                topic.canonical_story_references]
+            stories = story_fetchers.get_stories_by_ids(story_ids)
```

Comment:
This is still a get call in a for loop. Collate all the story IDs and get them all in one go outside the for loop.

Reply:
Done